lib/aarch64/parser.rb

```diff
@@ -218,10 +611,20 @@ def _next_token
         end
       elsif str = @scan.scan(SYS_REG_SCAN)
         [:SYSTEMREG, SYS_REG_MAP[str.downcase]]
-      elsif str = @scan.scan(/\w+/)
+      elsif str = @scan.scan(KEYWORDS_SCAN)
         [str.upcase.to_sym, str]
+      elsif str = @scan.scan(LABEL_SCAN)
+        [:LABEL, str]
       else
+        [:UNKNOWN_CHAR, @scan.getch]
       end
     end
+
+    def on_error(token_id, val, vstack)
```

Comment:
I found it hard to track down the parsing errors in the original input string when I was first playing with the gem; so I've tried to enhance this while I'm here. I can remove this change entirely or put it into a separate PR